File: app/src/main/java/com/example/notes/DescriptionFragment.java
package com.example.notes;

import android.app.DatePickerDialog;
import android.icu.util.Calendar;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.DatePicker;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.android.material.snackbar.Snackbar;

import java.util.Locale;


public class DescriptionFragment extends Fragment {

    private static final String ARG_ITEM = "param1";
    private static final String ARG_PARAM2 = "param2";
    public DataActions dataActions = new DataActionsImpl();
    private EditText editName;
    private EditText editDescription;
    private TextView dateTextField;
    private Button save;
    private DataNote mParam1;
    private String mParam2;
    private ListNotesFragment listNotesFragment = new ListNotesFragment();
    DataNote dataNote;


    Calendar dateAndTime = Calendar.getInstance(Locale.getDefault());
    DatePickerDialog.OnDateSetListener d = new DatePickerDialog.OnDateSetListener() {
        @Override
        public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
            dateAndTime.set(year, monthOfYear, dayOfMonth);
            dateAndTime.set(Calendar.MONTH, monthOfYear);
            dateAndTime.set(Calendar.DAY_OF_MONTH, dayOfMonth);
            android.text.format.DateFormat df = new android.text.format.DateFormat();
            String dateText = df.format("dd-MM-yyyy", dateAndTime.getTime()).toString();
            dateTextField.setText(dateText);
        }
    };

    public DescriptionFragment() {
        // Required empty public constructor
    }

    public static DescriptionFragment newInstance(DataNote dataNote1) {
        DescriptionFragment fragment = new DescriptionFragment();
        Bundle args = new Bundle();
        args.putParcelable(ARG_ITEM, dataNote1);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getParcelable(ARG_ITEM);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_description, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        editName = view.findViewById(R.id.name_edit_text);
        editDescription = view.findViewById(R.id.description_edit_text);
        dateTextField = view.findViewById(R.id.date_text_view);


        dateTextField.setOnClickListener(view12 -> {
            new DatePickerDialog(getContext(), d,
                    dateAndTime.get(Calendar.YEAR),
                    dateAndTime.get(Calendar.MONTH),
                    dateAndTime.get(Calendar.DAY_OF_MONTH))
                    .show();
        });

        save = view.findViewById(R.id.btn_save);
        save.setOnClickListener(view1 -> {


            if (dataActions != null) {

                DataNote data = new DataNote(mParam1.getId(), editName.getText().toString(),
                        editDescription.getText().toString(), dateTextField.getText().toString());
                dataActions.addListElement(data);

                getActivity().getSupportFragmentManager()
                        .beginTransaction()
                        .remove(this)
                        .commit();
            }
        });

        editName.setText(mParam1.getName());
        editDescription.setText(mParam1.getDescription());
        dateTextField.setText(mParam1.getDate());

    }

}

File: app/src/main/java/com/example/notes/DescriptionFragment.kt
package com.example.notes

import android.os.Bundle
import android.widget.TextView
import android.widget.EditText
import android.app.DatePickerDialog.OnDateSetListener
import android.app.DatePickerDialog
import android.icu.util.Calendar
import android.text.format.DateFormat
import android.view.*
import android.widget.Button
import androidx.core.view.get
import androidx.fragment.app.Fragment
import java.util.*

class DescriptionFragment : Fragment() {
    var dataActions: DataActions? = null
    private var editName: EditText? = null
    private var editDescription: EditText? = null
    private var dateTextField: TextView? = null
    private var save: Button? = null
    private var mParam1: DataNote? = null
    private var mParam2: String? = null
    var dateAndTime: Calendar = Calendar.getInstance(Locale.getDefault())
    var d: OnDateSetListener? = OnDateSetListener { _, year, monthOfYear, dayOfMonth ->
        dateAndTime[year, monthOfYear] = dayOfMonth
        dateAndTime[Calendar.MONTH] = monthOfYear
        dateAndTime[Calendar.DAY_OF_MONTH] = dayOfMonth
        val df = DateFormat()
        val dateText = DateFormat.format("dd-MM-yyyy", dateAndTime.time).toString()
        dateTextField?.text = dateText
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
        if (arguments != null) {
            mParam1 = arguments?.getParcelable(ARG_ITEM)
            mParam2 = arguments?.getString(ARG_PARAM2)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        val item: MenuItem = menu.findItem(R.id.action_clear)
        item.isVisible = false
        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        setHasOptionsMenu(true)
        return inflater.inflate(R.layout.fragment_description, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        editName = view.findViewById(R.id.name_edit_text)
        editDescription = view.findViewById(R.id.description_edit_text)
        dateTextField = view.findViewById(R.id.date_text_view)
        dateTextField?.setOnClickListener {
            context?.let {
                DatePickerDialog(
                    it, d,
                    dateAndTime[Calendar.YEAR],
                    dateAndTime[Calendar.MONTH],
                    dateAndTime[Calendar.DAY_OF_MONTH]
                )
                    .show()
            }
        }
        save = view.findViewById(R.id.btn_save)
        save?.setOnClickListener {
            if (dataActions != null) {
                val data = DataNote(
                    mParam1?.getId(),
                    editName?.text.toString(),
                    editDescription?.text.toString(),
                    dateTextField?.text.toString()
                )
                dataActions?.updateElement(data)
                activity?.supportFragmentManager
                    ?.beginTransaction()
                    ?.remove(this)
                    ?.commit()
            }
        }
        editName?.setText(mParam1?.getName())
        editDescription?.setText(mParam1?.getDescription())
        dateTextField?.text = mParam1?.getDate()
    }

    companion object {
        private val ARG_ITEM: String? = "param1"
        private val ARG_PARAM2: String? = "param2"
        fun newInstance(dataNote1: DataNote?): DescriptionFragment? {
            val fragment = DescriptionFragment()
            val args = Bundle()
            args.putParcelable(ARG_ITEM, dataNote1)
            fragment.arguments = args
            return fragment
        }
    }
}